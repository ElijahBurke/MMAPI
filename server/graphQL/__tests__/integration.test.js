const { createTestClient } = require('apollo-server-testing');
const { ApolloServer } = require('apollo-server');
const gql = require('graphql-tag');
const mocks = require('../../express/mocks');

const db = require('../../express/models');
const typeDefs = require('../schemas');
const resolvers = require('../resolvers');

const server = new ApolloServer({
  typeDefs,
  resolvers,
});

const GET_FIGHTS = gql`
query fights {
  fights {
    id
    winner
    method
  }
}
`;

const GET_FIGHT = gql`
query fights {
  fights(id: 1) {
    id
    winner
    method
  }
}
`;

describe('Queries', () => {
  const { query } = createTestClient(server);
  let result;

  beforeAll(async () => {
    await db.sequelize.sync({ force: true });
    await db.fights.bulkCreate([mocks.fightOne, mocks.fightTwo]);
  });

  afterAll(async () => {
    await db.sequelize.drop();
    await db.sequelize.close();
  });

  test('Gets all fights', async () => {
    result = await query({ query: GET_FIGHTS });
    expect(result.data).toMatchSnapshot();
    expect(1).toBe(1);
  });
  test('Gets correct fight', async () => {
    result = await query({ query: GET_FIGHT });
    expect(result.data.fights[0].id).toBe(1);
    // expect(1).toBe(1);
  });
});
