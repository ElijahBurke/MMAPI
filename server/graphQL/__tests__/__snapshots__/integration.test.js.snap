// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries Gets all fights 1`] = `
Object {
  "fights": Array [
    Object {
      "id": 1,
      "method": "fighting",
      "winner": "red",
    },
    Object {
      "id": 2,
      "method": "screaming",
      "winner": "blue",
    },
  ],
}
`;
